src/general/Filter.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+
+// TODO CONSIDER GENERIC
+public sealed class Filter<T> : IFilter
+{
+    string IFilter.FilterCategory
+    {
+        get => filterCategory;
+        set => filterCategory = value;
+    }
+
+    private string filterCategory = "NONE";
+    //public string FilterCategory = "NONE";
+
+    private Dictionary<string, IFilter.IFilterItem> filterItems = new Dictionary<string, IFilter.IFilterItem>();
+
+    public IEnumerable<string> FilterItemsNames => filterItems.Keys;
+
+    Dictionary<string, IFilter.IFilterItem> IFilter.FilterItems => filterItems;
+
+    public void AddFilterItem(string category, FilterItem item)
+    {
+        filterItems.Add(category, item);
+    }
+
+    public void ClearItems()
+    {
+        filterItems.Clear();
+    }
+
+    public Func<T, bool> ComputeFilterFunction()
+    {
+        if (!filterItems.TryGetValue(filterCategory, out var filterItem))
```

Comment:
If you are to throw a KeyNotFoundException, why don't you use `filterItems[filterCategory]` instead?